pkg/alertmanager/amcfg.go

```diff
@@ -390,6 +452,152 @@ func (cg *configGenerator) convertWebhookConfig(ctx context.Context, in monitori
 	return out, nil
 }
 
+func (cg *configGenerator) convertSlackConfig(ctx context.Context, in monitoringv1alpha1.SlackConfig, crKey types.NamespacedName) (*slackConfig, error) {
+	out := &slackConfig{}
+
+	if in.SendResolved != nil {
+		out.VSendResolved = *in.SendResolved
+	}
+
+	if in.APIURL != nil {
+		url, err := cg.store.GetSecretKey(ctx, crKey.Namespace, *in.APIURL)
+		if err != nil {
+			return nil, errors.Errorf("failed to get key %q from secret %q", in.APIURL.Key, in.APIURL.Name)
+		}
+		out.APIURL = url
+	}
+
+	if in.Channel != nil {
+		out.Channel = *in.Channel
+	}
+
+	if in.Username != nil {
+		out.Username = *in.Username
+	}
+
+	if in.Color != nil {
+		out.Color = *in.Color
+	}
+
+	if in.Title != nil {
+		out.Title = *in.Title
+	}
+
+	if in.TitleLink != nil {
+		out.TitleLink = *in.TitleLink
+	}
+
+	if in.Pretext != nil {
+		out.Pretext = *in.Pretext
+	}
+
+	if in.Text != nil {
+		out.Text = *in.Text
+	}
+
+	if in.ShortFields != nil {
+		out.ShortFields = *in.ShortFields
+	}
+
+	if in.Footer != nil {
+		out.Footer = *in.Footer
+	}
+
+	if in.Fallback != nil {
+		out.Fallback = *in.Fallback
+	}
+
+	if in.CallbackID != nil {
+		out.CallbackID = *in.CallbackID
+	}
+
+	if in.IconEmoji != nil {
+		out.IconEmoji = *in.IconEmoji
+	}
+
+	if in.IconURL != nil {
+		out.IconURL = *in.IconURL
+	}
+
+	if in.ImageURL != nil {
+		out.ImageURL = *in.ImageURL
+	}
+
+	if in.ThumbURL != nil {
+		out.ThumbURL = *in.ThumbURL
+	}
+
+	if in.LinkNames != nil {
+		out.LinkNames = *in.LinkNames
+	}
+
+	out.MrkdwnIn = in.MrkdwnIn
+
+	var actions []slackAction
+	if len(in.Actions) > 0 {
+		actions = make([]slackAction, 0)
```

Comment:
Suggestion for these kinds of loops: I think this will save some computational effort, given that you know wht the size of the slice will be upfront. This might mean that you'd need to say e.g. `actions[i] = action`, rather than `append(actions, action)` later though.

```suggestion
	if l := len(in.Actions); l > 0 {
		actions = make([]slackAction, l)
```

Reply:
Thanks :)